src/store.ts

```diff
@@ -50,6 +50,8 @@ export class Store {
   /**
    * Selects a slice of data from the store.
    */
+  select<T>(selector: (state: any) => T): Observable<T>;
```

Comment:
what if instead of the return type, "T" was the state and we remove the return signature so the compiler could infer the correct type.

```
select<T>(selector: (state: T) => T);
```